src/Api/Controller/AbstractSerializeController.php

```diff
@@ -194,6 +220,27 @@ protected function loadRelations(Collection $models, array $relations): void
 
         if (! empty($relations)) {
             $relations = array_unique($relations);
+        }
+
+        $callables = [];
+
+        foreach ($relations as $k => $relation) {
```

Comment:
It's been a bit challenging to understand this. Instead of unsetting `$relations`, could we have `callableRelations` and `nonCallableRelations`, and generate both from `relations`? That way, it's a computational transformation and not mutation-based.